sentinelhub/data_utils.py

```diff
@@ -116,37 +126,46 @@ def statistical_to_dataframe(result_data: List[JsonDict], exclude_stats: Optiona
     return pandas.concat(dfs)
 
 
-def _get_failed_intervals(
-    identifier: str, response_data: List[JsonDict]
-) -> Optional[Dict[str, Union[str, List[Tuple[str, str]]]]]:
-    """Collect failed intervals of a partially failed request
+def _get_failed_intervals(result_data: JsonDict) -> Union[str, List[Tuple[str, str]]]:
+    """Collect failed intervals of a single geometry from the (Batch) Statistical result
 
-    :param identifier: The identifier of the geometry.
-    :param response_data: An input representation of Statistical API response.
-    :return: The identifier of a geometry that has a response status of PARTIAL and the failed intervals.
+    :param result_data: An input representation of the (Batch) Statistical API result of a geometry.
+    :return: The failed intervals for a geometry.
     """
-    failed_intervals = []
-    for interval in response_data:
-        if "error" in interval:
-            failed_intervals.append((interval["interval"]["from"], interval["interval"]["to"]))
-
-    return {"identifier": identifier, "failed_intervals": failed_intervals} if failed_intervals else None
-
-
-def get_failed_statistical_requests(result_data: List[JsonDict]) -> List[Dict[str, Union[str, List[Tuple[str, str]]]]]:
+    if _is_batch_stat(result_data):
+        return (
+            "full time range"
+            if "error" in result_data or not result_data["response"]
+            else [
+                (interval["interval"]["from"], interval["interval"]["to"])
+                for interval in result_data["response"]["data"]
+                if "error" in interval
+            ]
+        )
```

Comment:
this kind of nesting is very hard to read in python. Move the conditional out of the `return` statement. 

You also use `"full time range"` in many places, so you should extract it as a constant